Lib/posixpath.py

```diff
@@ -467,38 +475,46 @@ def realpath(filename, *, strict=False):
         else:
             newpath = path + sep + name
         try:
-            st = os.lstat(newpath)
+            st = lstat(newpath)
             if not stat.S_ISLNK(st.st_mode):
                 path = newpath
                 continue
-            if newpath in seen:
+            elif maxlinks is not None:
+                link_count += 1
+                if link_count > maxlinks:
+                    if strict:
+                        raise OSError(errno.ELOOP, "Too many symbolic links in path", newpath)
```

Comment:
Use macOS error message:

```suggestion
                        raise OSError(errno.ELOOP, "Too many levels of symbolic links", newpath)
```